### ISSUE TITLE
When remixing, if a tutorial is present, show it in the preview by default.

### ISSUE THREAD

**Issue by flukeout** [mozilla/thimble.mozilla.org#1776]
The projects that we'll feature on the homepage all have tutorials. However, when you remix one of them, the tutorial isn't displayed prominently, and there's a good chance it might get missed.

If possible, when remixing, can we show the tutorial in the preview pane if it's available? This means that in the header, the Tutorial option should be enabled like this...

![image](https://cloud.githubusercontent.com/assets/25212/23237153/fa5acf48-f910-11e6-9c3a-238a1598ae22.png)

In the editor view, however, we should still show whatever the default file logic is (first HTML file in the list I think?)

**Comment by Pomax** [comment]
should things like reloads etc still show when preview is selected? it kind of feels like alpha'ing them to 0.1 or something and making them inaccessible would be necessary if we're going to present the option in the preview pane

**Comment by babhishek21** [comment]
@flukeout So right after clicking the "Remix this Project" button, when the bramble panes load, you want the Tutorial pane being displayed by default, instead of the Preview pane? 

Also, can I work on this?

**Comment by humphd** [comment]
@babhishek21 yes, you can work on this.  I suspect that the right fix for this is to alter our code here:

https://github.com/mozilla/thimble.mozilla.org/blob/392fb9dccc4a228b21dfe88cfe415cb56a6acded/public/editor/scripts/project/project.js#L240-L257

This is where we decide which file to show first when we load a project.  If there's a `tutorial.html` file, we should favour that over something else.

**Comment by humphd** [comment]
From Mattermost:

> babhishek21: @humphd Regarding issue #1776: As I suspected, changing public/editor/scripts/project/project.js#L240-L257 only solves the problem partially. This is what happens on loading/remixing a project with an existing "tutorial.html". The Preview and Editor panes indeed show the contents of the tutorial. However, the Tutorial pane should've be activated instead of the Preview pane. That isn't happening. 


![image](https://cloud.githubusercontent.com/assets/427398/23464313/a1851020-fe62-11e6-8791-d6d3fed621a5.png)

My response:

> humphd: @babhishek21 you'll want to trigger an event, tutorialVisibilityChange, when you load tutorial.html probably in here https://github.com/mozilla/brackets/blob/master/src/extensions/default/bramble/main.js#L128 the code you want to call is in src/bramble/BrambleEvents.js
```

**Comment by humphd** [comment]
@babhishek21 are you still working on this?  If not, can you let us know so we can put it back in the general pool of bugs for people to take?  Thanks.

**Comment by babhishek21** [comment]
Yup, still working on this! I'm sorry, I was on vacation last week and did not have access to my devbox (I wrote that on Mattermost, but it must have been lost in the noise) 😓 Expect a patch by today evening or tomorrow morning.

**Comment by babhishek21** [comment]
I did what @humphd suggested. It didn't seem to work. 😭 

So I did the following:
+ Import the `BrambleEvents module
```js
var BrambleEvents  = brackets.getModule("bramble/BrambleEvents")
```
+ Fire off an event to change tutorial pane visibility [here](https://github.com/mozilla/brackets/blob/master/src/extensions/default/bramble/main.js#L145-L164)
```js
BrambleEvents.triggerTutorialVisibilityChange(filename.endsWith('tutorial.html')
// BrambleEvents.triggerTutorialVisibilityChange(true)                   // tried this as well
// BrambleEvents.triggerTutorialVisibilityChange(false)                   // tried this as well
```

And while these lines execute without errors (I checked with a couple of `console.log()`s), the tutorial pane still isn't activated. It is always showing the preview pane.

It is possible that after the event to activate the tutorial pane is fired off
1. there is no function to handle that event.    **OR**
2. another function or event fired off later, forces the preview pane to be activated.

**Comment by humphd** [comment]
You're sending that event too early.  Thimble and Brackets aren't setup to talk to each other until after `RemoteEvents.loaded()` completes, see https://github.com/mozilla/brackets/blob/master/src/extensions/default/bramble/lib/RemoteEvents.js#L161-L194.

You probably want to trigger your event after this completes https://github.com/mozilla/brackets/blob/master/src/extensions/default/bramble/main.js#L125

**Comment by babhishek21** [comment]
OK I did a bit of a looky-loo. Take a look at the inspector messages and the window in this image:
![image](https://cloud.githubusercontent.com/assets/8472087/24497673/d1fd5da8-1559-11e7-8540-a07ec31c6089.png)


To give you bit of a background: 
+ The `tutorial event activatedtrue` message means that I triggered a tutorial visibility change using:
```js
BrambleEvents.triggerTutorialVisibilityChange(true); 
```
+ The `Loaded() is done` message means `RemoteEvents.loaded()` (see [here](https://github.com/mozilla/brackets/blob/master/src/extensions/default/bramble/lib/RemoteEvents.js#L161-L194) is done.
+ The `Loaded() has returned!` message means [this line](https://github.com/mozilla/brackets/blob/master/src/extensions/default/bramble/main.js#L125) finished execution.

**Comment by babhishek21** [comment]
Feel free to safely ignore the contents of the [previous comment.](https://github.com/mozilla/thimble.mozilla.org/issues/1776#issuecomment-290357529) 

Looks like a small typo in Thimble client side was preventing the `bramble:tutorialVisibilityChange` event from being interpreted properly. As to how it was working before this is honestly beyond me. 

The repective PR(s) are up: Thimble #2001 and Bramble [#722](https://github.com/mozilla/brackets/issues/722)

From my time debugging this, I am not entirely sure the state stored in the `bramble` object in the browser scope passes integerity concerns.
